cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java

```diff
@@ -14,62 +15,152 @@
-    private Integer cpuUtilization;
-    private Double inboundNetworkKiBPerSecond;
-    private Double outboundNetworkKiBPerSecond;
+    protected static final ReconciliationLogger LOGGER = ReconciliationLogger.create(Capacity.class.getName());
+
+    private TreeMap<Integer, Broker> capacityEntries;
 
     private int replicas;
     private Storage storage;
 
     public Capacity(KafkaSpec spec, Storage storage) {
         BrokerCapacity bc = spec.getCruiseControl().getBrokerCapacity();
 
+        this.capacityEntries = new TreeMap<Integer, Broker>();
         this.replicas = spec.getKafka().getReplicas();
         this.storage = storage;
 
-        this.diskMiB = generateDiskCapacity(storage);
-        this.cpuUtilization = DEFAULT_BROKER_CPU_UTILIZATION_CAPACITY;
-        this.inboundNetworkKiBPerSecond = bc != null && bc.getInboundNetwork() != null ? getThroughputInKiB(bc.getInboundNetwork()) : DEFAULT_BROKER_INBOUND_NETWORK_CAPACITY_IN_KIB_PER_SECOND;
-        this.outboundNetworkKiBPerSecond = bc != null && bc.getOutboundNetwork() != null ? getThroughputInKiB(bc.getOutboundNetwork()) : DEFAULT_BROKER_OUTBOUND_NETWORK_CAPACITY_IN_KIB_PER_SECOND;
+        processCapacityEntries(bc, storage);
     }
 
-    /**
-     * Generate broker capacity entry for capacity configuration.
-     *
-     * @param brokerId Id of broker
-     * @param diskCapacity Disk capacity configuration
-     * @param doc Documentation for broker entry
-     * @return Broker entry as a JsonObject
-     */
-    private JsonObject generateBrokerCapacity(int brokerId, JsonObject diskCapacity, String doc) {
-        JsonObject brokerCapacity = new JsonObject()
-            .put("brokerId", brokerId)
-            .put("capacity", new JsonObject()
-                .put("DISK", diskCapacity.getValue("DISK"))
-                .put("CPU", Integer.toString(cpuUtilization))
-                .put("NW_IN", Double.toString(inboundNetworkKiBPerSecond))
-                .put("NW_OUT", Double.toString(outboundNetworkKiBPerSecond))
-            )
-            .put("doc", doc);
-        return brokerCapacity;
+    public static String processCpu() {
+        return Broker.DEFAULT_CPU_UTILIZATION_CAPACITY;
+    }
+
+    public static String processDisk(Storage storage, int brokerId) {
+        if (storage instanceof  JbodStorage) {
+            return generateJbodDiskCapacity(storage, brokerId).encodePrettily();
+        } else {
+            return generateDiskCapacity(storage);
+        }
+    }
+
+    public static String processInboundNetwork(BrokerCapacity bc, BrokerCapacityOverride override) {
+        if (override != null && override.getInboundNetwork() != null) {
+            return getThroughputInKiB(override.getInboundNetwork());
+        } else if (bc != null && bc.getInboundNetwork() != null) {
+            return getThroughputInKiB(bc.getInboundNetwork());
+        } else {
+            return Broker.DEFAULT_INBOUND_NETWORK_CAPACITY_IN_KIB_PER_SECOND;
+        }
```

Comment:
Do you need the `else` block here? It's just a nit, but I think you can do 
```suggestion
        if (override != null && override.getInboundNetwork() != null) {
            return getThroughputInKiB(override.getInboundNetwork());
        } else if (bc != null && bc.getInboundNetwork() != null) {
            return getThroughputInKiB(bc.getInboundNetwork());
        }
        return Broker.DEFAULT_INBOUND_NETWORK_CAPACITY_IN_KIB_PER_SECOND;
```

maybe same for the `processOutboundNetwork`

Reply:
We don't need the `else` block here but I think it is good to list it explicitly to help avoid any possible confusion

Reply:
Okay, makes sense :)